core/src/main/scala/kafka/server/FinalizedFeatureCache.scala

```diff
@@ -39,7 +42,7 @@ case class FinalizedFeaturesAndEpoch(features: Features[FinalizedVersionRange],
  *
  * @see FinalizedFeatureChangeListener
  */
-object FinalizedFeatureCache extends Logging {
+class FinalizedFeatureCache(private val brokerFeatures: BrokerFeatures) extends Logging {
```

Comment:
The meta comment for `FinalizedFeatureCache` should be updated as it is now being accessed for both read and write